apps/examples/smartfs/read_optimization/read_optimization_main.c

```diff
@@ -0,0 +1,273 @@
+	}
+
+	frt_fd = open(timer_path, O_RDONLY);
+	if (frt_fd < 0) {
+		fprintf(stderr, "ERROR: Failed to open Free Run Timer: %d\n", errno);
+		return -1;
+	}
+
+	if (ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN) < 0) {
+		fprintf(stderr, "ERROR: Failed to set Free Run Timer: %d\n", errno);
+		return -1;
+	}
+	if (ioctl(frt_fd, TCIOC_START, TRUE) < 0) {
+		fprintf(stderr, "ERROR: Failed to start Free Run Timer: %d\n", errno);
+		return -1;
+	}
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_1) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	read_test(512);
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_2) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	time_read = time_2.timeleft - time_1.timeleft;
+	printf("\n****************************************************************************************\n");
+	printf("Time taken to read %dKb file with 512 byte buffer, 100 times = %lu\n", size, time_read);
+	printf("\n****************************************************************************************\n");
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_1) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	read_test(1024);
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_2) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	time_read = time_2.timeleft - time_1.timeleft;
+	printf("\n****************************************************************************************\n");
+	printf("Time taken to read %dKb file with 1024 byte buffer, 100 times = %lu\n", size, time_read);
+	printf("\n****************************************************************************************\n");
+
+	ret = stat("/mnt", &st);
+	if (ret < 0) {
+		printf("Stat to get optimal read buffer size failed, Errno = %d\n", errno);
+		return -1;
+	}
+	buf_size = st.st_blksize;
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_1) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	read_test(buf_size);
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_2) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	time_read = time_2.timeleft - time_1.timeleft;
+	printf("\n****************************************************************************************\n");
+	printf("Time taken to read %dKb file with %d byte buffer, 100 times = %lu\n", size, buf_size, time_read);
+	printf("\n****************************************************************************************\n");
+
+	if (ioctl(frt_fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the Free Run Timer: %d\n", errno);
+	}
+
+	unlink(SAMPLE_FILE);
```

Comment:
Can we add a check here also please whether or not unlink fails ?

Reply:
We can work without it since it's only a sample application here.